packages/devtools/lib/src/popup.dart

```diff
@@ -0,0 +1,632 @@
+  }
+
+  /// Returns the row number of item to make visible.
+  int page(ListDirection direction, [int startRow = 0]) {
+    final int listHeight = element.element.clientHeight;
+    final int itemHeight = items.element.children[0].clientHeight;
+    final int itemsVis = (listHeight / itemHeight).truncate() - 1;
+
+    int childToScrollTo;
+    switch (direction) {
+      case ListDirection.pageDown:
+        int itemIndex = startRow + itemsVis;
+        if (itemIndex > items.items.length - 1) {
+          itemIndex = items.items.length - 1;
+        }
+        childToScrollTo = itemIndex;
+        final int scrollPosition = startRow > 0 ? startRow * itemHeight : 0;
+        scrollAndHilight(childToScrollTo, scrollPosition, top: true);
+        break;
+      case ListDirection.pageUp:
+        int itemIndex = startRow - itemsVis;
+        if (itemIndex < 0) itemIndex = 0;
+        childToScrollTo = itemIndex;
+        final int scrollPosition =
+            childToScrollTo > 0 ? childToScrollTo * itemHeight : 0;
+        scrollAndHilight(childToScrollTo, scrollPosition, top: true);
+        break;
+      case ListDirection.home:
+        childToScrollTo = 0;
+        scrollAndHilight(childToScrollTo, childToScrollTo);
+        break;
+      case ListDirection.end:
+        childToScrollTo = items.items.length - 1;
+        final int scrollPosition =
+            childToScrollTo > 0 ? (childToScrollTo - itemsVis) * itemHeight : 0;
+        scrollAndHilight(childToScrollTo, scrollPosition);
+        break;
+    }
+
+    return childToScrollTo;
+  }
+
+  SelectableList<T> items;
+  T highlightedItem;
+
+  List<T> get itemsAsList => items.items;
+  bool get isEmpty => items.items.isEmpty;
+
+  @override
+  CoreElement get element => items;
+
+  bool get itemsHadClicked => items.hadClicked;
+
+  Stream<html.KeyboardEvent> get onKeyDown {
+    return items.onKeyDown;
+  }
+
+  Stream<T> get onSelectionChanged {
+    return items.onSelectionChanged;
+  }
+
+  Stream<void> get onScriptsChanged {
+    return items.onItemsChanged;
+  }
+
+  void setList(
+    List<T> theItems, {
+    T select,
+  }) {
+    theItems.sort((T item1, T item2) {
+      return item1.toString().compareTo(item2.toString());
+    });
+
+    T selection;
+    if (select != null) {
+      selection = select;
+    }
+
+    items.setItems(theItems,
+        selection: selection, scrollSelectionIntoView: true);
+  }
+
+  void clearList() => items.clearItems();
+}
+
+/// This is the view that manages Input element (popupTextField) and the popup
+/// list being displayed (_listView).  Calling show() displays the popup list
+/// directly below the Input element (popupTextfield).
+class PopupAutoCompleteView extends CoreElement {
+  PopupAutoCompleteView(
+    this._listView,
+    this._containerElement,
+    this._popupTextfield,
+    this._completeAction,
+  ) : super('div', classes: 'popup-view') {
+    _initialize();
+  }
+
+  PopupAutoCompleteView.mimic(
```

Comment:
perhaps add dartdoc here to describe what mimic means

Reply:
Done.